src/processes/helper/sendTopicAutoMessage.js

```diff
@@ -4,21 +4,29 @@ const {sample} = require('lodash');
 const {followTopicQueue} = require('../../config/redis');
 const {CommunityMessageFormat, LogError} = require('../../databases/models');
 
+const setRequiredTime = (delay) => {
+  let requiredTime = momentTz().tz('America/Los_Angeles');
+  if (delay === 0) {
+    const randomTime = Math.floor(Math.random() * (40 - 10 + 1)) + 10;
+    requiredTime = momentTz().tz('America/Los_Angeles').add(randomTime, 'minutes');
+  } else {
+    const randomTime = sample([6, 7, 8]);
+    const additionalDays = delay;
+    requiredTime = momentTz()
+      .tz('America/Los_Angeles')
+      .set({hour: randomTime})
+      .add(additionalDays, 'days');
+  }
+  return requiredTime;
+};
 const followTopicServiceQueue = async (userId, topicId, communityMessageFormatId, delay) => {
   const data = {
     user_id: userId,
     topic_id: topicId,
     community_message_format_id: communityMessageFormatId
   };
   const currentTime = momentTz().tz('America/Los_Angeles');
-  const randomTime = sample([6, 7, 8]);
-  const additionalDays = delay;
-
-  const requiredTime = momentTz()
-    .tz('America/Los_Angeles')
-    .set({hour: randomTime})
-    .add(additionalDays, 'days');
-
+  const requiredTime = setRequiredTime(delay);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `followTopicServiceQueue` function is missing its closing bracket and return statement. Make sure to close all functions properly and return the necessary data. Also, ensure that the `options` object is being used somewhere in the function.

```diff
  const requiredTime = setRequiredTime(delay);
  const diffTime = requiredTime.diff(currentTime, 'milliseconds');

  const options = {
    // your options here
  };

  // use options here

  return something; // return the necessary data
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->